### DISCUSSION TITLE
CLI command to set a new wallpaper

### DISCUSSION THREAD

**Discussion by PogopunkXIII** [Jas-SinghFSU/HyprPanel#484] [Q&A] [answered]
I have a directory of wallpapers and I have a script that runs on startup and selects a wallpaper to display. I'd like to be able to add a command to the script to set the wallpaper via hyprpanel so I can get the effect of the matugen color generation.

**Comment by Jas-SinghFSU** [comment, marked as answer]
You can use the wallpaper cli command listed here: https://hyprpanel.com/configuration/cli.html#utility-commands

**Comment by PogopunkXIII** [comment]
ahh I am so blind, I literally read that page and just missed the command, thanks.

**Comment by PogopunkXIII** [comment]
when I try running `ags -r "setWallpaper('<PATH>')"` I get `ReferenceError: setWallpaper is not defined`

**Reply by Jas-SinghFSU** [reply]
Can you share the output of this command please:
```
git rev-parse HEAD
```

You may not be on latest branch.

**Reply by PogopunkXIII** [reply]
output: `a7855baf13c6abdd0b0e988e4390112cd7deda67`

**Reply by Jas-SinghFSU** [reply]
Oh wait are you on nix?

**Reply by PogopunkXIII** [reply]
nope, arch

**Reply by Jas-SinghFSU** [reply]
Odd, I'm literally running that command right now lol. Did you recently update? If so you may have to restart it.

**Reply by PogopunkXIII** [reply]
ahh I did just pull during this conversation, I did not restart though, let me do that

**Reply by PogopunkXIII** [reply]
that was it, all set now, thanks a ton

**Reply by Jas-SinghFSU** [reply]
Np 😊
